modules/core/src/test/java/org/apache/ignite/cache/affinity/AffinityHistoryCleanupTest.java

```diff
@@ -224,6 +228,54 @@ private void checkHistory(Ignite ignite, List<AffinityTopologyVersion> expHist,
         assert cnt > 4;
     }
 
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    @WithSystemProperty(key = IGNITE_AFFINITY_HISTORY_SIZE, value = "10")
+    @WithSystemProperty(key = IGNITE_MIN_AFFINITY_HISTORY_SIZE, value = "2")
+    public void testOldNonShallowAffinityHistoryIsRemoved() throws Exception {
+        testNonShallowAffinityHistory(true);
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    @WithSystemProperty(key = IGNITE_AFFINITY_HISTORY_SIZE, value = "10")
+    @WithSystemProperty(key = IGNITE_MIN_AFFINITY_HISTORY_SIZE, value = "3")
+    public void testNonShallowAffinityHistoryIsNotRemoved() throws Exception {
+        testNonShallowAffinityHistory(false);
+    }
+
+    private void testNonShallowAffinityHistory(boolean isRemoved) throws Exception {
+        IgniteEx ignite = startGrids(2);
+
+        ignite.createCache(new CacheConfiguration<>(DEFAULT_CACHE_NAME)
+            .setAffinity(new RendezvousAffinityFunction(false, 1)));
+
+        GridCacheProcessor proc = ignite.context().cache();
+
+        GridCacheContext<Object, Object> cctx = proc.context().cacheContext(CU.cacheId(DEFAULT_CACHE_NAME));
+
+        GridAffinityAssignmentCache aff = GridTestUtils.getFieldValue(cctx.affinity(), "aff");
```

Comment:
Honestly, I don't like using reflection in this case. Is it possible to change the test as follows:
```
    GridCacheContext<Object, Object> cctx = proc.context().cacheContext(CU.cacheId(DEFAULT_CACHE_NAME));

    GridCacheAffinityManager affMgr = cctx.affinity();
    AffinityTopologyVersion topVer = affMgr.affinityTopologyVersion();
    ...
        if (isRemoved)
        assertThrowsWithCause(() -> affMgr.nodesByPartition(0, topVer), IllegalStateException.class);
    else
        assertEquals(1, affMgr.nodesByPartition(0, topVer).size());
```

WDYT?

Reply:
Agree, fixed.